client/components/search/index.jsx

```diff
@@ -201,15 +198,11 @@ const Search = React.createClass( {
 	},
 
 	keyUp: function( event ) {
-		if ( event.which === 13 && isMobile() ) {
+		if ( event.key === 'Enter' && isMobile() ) {
```

Comment:
keyboard events are so fun. 

should probably check for key and use it if it's there, otherwise fallback to which.

Is react doing any normalization of the event?